### ISSUE TITLE
memory error - most likely due to large data set

### ISSUE THREAD

**Issue by directgumby** [mathuin/TopoMC#34]
Hello,

I am attempting to make a rather large world from this data set.  Here is my error:

Traceback (most recent call last):
  File ".\prepregion.py", line 28, in <module>
    sys.exit(main())
  File ".\prepregion.py", line 25, in main
    myRegion.build_map(args.doOCL, args.doPickle)
  File "C:\Users\Brett\Documents\GitHub\TopoMC\region.py", line 494, in build_map
    crustarray = newcrust(pickle_name=pickle_name)
  File "C:\Users\Brett\Documents\GitHub\TopoMC\crust.py", line 28, in **call**
    retval = self.idt(self.base, self.shape, majority=False, pickle_name=pickle_name)
  File "C:\Users\Brett\Documents\GitHub\TopoMC\idt.py", line 141, in **call**
    distances, indexes = self.tree.query(base, k=nnear)
  File "ckdtree.pyx", line 1227, in scipy.spatial.ckdtree.cKDTree.query (scipy\spatial\ckdtree.c:10464)
MemoryError

Here is the map I was attempting to build:
python .\getregion.py --name Flathead --ymin 47.6765 --ymax 48.2211 --xmin -114.0096 --xmax -114.4545 --vscale 14 --enable-schematics

Do you have any suggestions on ways to avoid this and/or work around it in a way that can give a good smooth representation of that area?

Thank you,
Brett

**Comment by mathuin** [comment]
MemoryError usually means that you've run out of RAM.  I would recommend that you consider increasing your virtual memory.  On Linux this means adding a swapfile.  I see that you're running Windows but I don't know exactly what version.  The instructions to do this for Vista are here: http://windows.microsoft.com/en-us/windows-vista/change-the-size-of-virtual-memory  I imagine they're pretty much the same from operating system to operating system.  Your region isn't too huge -- how much RAM do you have?

Jack.

**Comment by directgumby** [comment]
I am running Windows 8, 4 gigs of ram, swap file is system set to be up to 10 gigs, and I have 45 gigs free on the drive.  I can expand this if needed but that seems sufficient.

A few more notes:

Pythonversion 2.7

Scipy 0.12.0

Pay-yaml 3.10

Pywin32 218.4

Numpy 1.7.1

Gadal 1.10.0

All are 64 bit.

Any other suggestions?

Thanks,

Brett

Sent from Windows Mail

From: Jack Twilley
Sent: ‎Monday‎, ‎August‎ ‎19‎, ‎2013 ‎7‎:‎29‎ ‎PM
To: mathuin/TopoMC
Cc: directgumby

MemoryError usually means that you've run out of RAM. I would recommend that you consider increasing your virtual memory. On Linux this means adding a swapfile. I see that you're running Windows but I don't know exactly what version. The instructions to do this for Vista are here: http://windows.microsoft.com/en-us/windows-vista/change-the-size-of-virtual-memory I imagine they're pretty much the same from operating system to operating system. Your region isn't too huge -- how much RAM do you have?

Jack.

**Comment by directgumby** [comment]
Hello,

I did two things:
1) doubled the cache 
2) deleted all of the usgs data and reran the getregion

I ended up with the same results - I have upped my cache to 40 gigs but I suspect there is something else going on here. :) 

Another note: I am not using (that I know of) the compiled bins that need to be created so I am using the lower performance part of some of the code.  Could this be part of the issue?

If you have any other suggestions I am happy to try them out.

Thanks,
Brett
